src/workerd/api/http.c++

```diff
@@ -1595,6 +1595,23 @@ jsg::Promise<jsg::Ref<Response>> handleHttpRedirectResponse(jsg::Lock& js,
     return js.rejectedPromise<jsg::Ref<Response>>(kj::mv(exception));
   }
 
+
+  if (FeatureFlags::get(js).getStripAuthorizationOnCrossOriginRedirect()) {
+    auto base = urlList.back().toString();
+    auto currentUrl = jsg::Url::tryParse(base.asPtr());
+    auto locationUrl = jsg::Url::tryParse(location, base.asPtr());
+    if (KJ_ASSERT_NONNULL(currentUrl).getOrigin() != KJ_ASSERT_NONNULL(locationUrl).getOrigin()) {
```

Comment:
It is entirely possible for either one of these to fail, in which case the `KJ_ASSERT_NONNULL` calls below would result in an `internal error` being thrown. We should check to see what browsers do when the URL returned by the location header is invalid and cannot be parsed. Does it just ignore it? Does it go ahead and remove the `authorization` header? etc.

If it matches the behavior in browsers, then what I would suggest here is that if either of these fail parsing, we just go ahead and remove the `authorization` header and do not throw any errors.

Reply:
Try this one... I know chrome doesn't like it: `'ÿ: not a valid url'`

Reply:
the address line will always try to be as resilient as possible. Just throw garbage into the `Location` head and see what happens.